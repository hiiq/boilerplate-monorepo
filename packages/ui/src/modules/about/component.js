import React from 'react';
import { Body, Content, Footer, Header, Breadcrumbs } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';

const About = () => {
  const t = useTranslate({
    component: 'about',
    namespace: 'about',
  });

  return (
    <Content>
      <Breadcrumbs />
      <Header title={t('title')} />
      <Body>{t('body')}</Body>
      <Footer>{t('footer')}</Footer>
    </Content>
  );
};

export { About };
