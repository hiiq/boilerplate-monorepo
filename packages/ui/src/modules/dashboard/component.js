import React from 'react';
import { Body, Content, Footer, Header, Breadcrumbs } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';

const Dashboard = () => {
  const t = useTranslate({
    component: 'dashboard',
    namespace: 'dashboard',
  });

  return (
    <Content>
      <Breadcrumbs />
      <Header title={t('title')} />
      <Body>
        <div
          dangerouslySetInnerHTML={{
            __html: t('body'),
          }}
        />
      </Body>
      <Footer>{t('footer')}</Footer>
    </Content>
  );
};

export { Dashboard };
