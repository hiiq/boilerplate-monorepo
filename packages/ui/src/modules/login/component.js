import { useMutation } from '@apollo/client';
import React, { useState } from 'react';
import { Button as SharedButton, Context } from 'shared/Button';
import { Body, Breadcrumb, Breadcrumbs, Content, Header } from 'shared/Content';
import { Link } from 'shared/Link';
import { Page } from 'shared/Page';
import { Mutation } from 'shared/graphql/mutation';
import { Query } from 'shared/graphql/query';
import { useAuth } from 'shared/useAuth';
import { useTranslate } from 'shared/useTranslate';
import styled from 'styled-components/macro';
import { Route } from 'types/route';

const { PRIMARY } = Context;

const Button = styled(SharedButton)`
  width: fit-content;
`;

// eslint-disable-next-line max-statements
const Login = () => {
  const t = useTranslate({
    component: 'login',
    namespace: 'login',
  });

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const { isAuthenticated, logout, updateAccessToken } = useAuth();

  const [mutate, { loading: isDisabled }] = useMutation(Mutation.LOGIN, {
    refetchQueries: [{ query: Query.ME }],
  });

  const onLogin = async () => {
    const input = {
      password,
      username,
    };

    await mutate({
      update: (_cache, { data }) => {
        const { userLogin: accessToken } = data;

        if (!accessToken) {
          logout();

          return;
        }

        updateAccessToken(accessToken);
      },
      variables: { input },
    });
  };

  const onUsernameChange = e => {
    setUsername(e.target.value);
  };

  const onPasswordChange = e => {
    setPassword(e.target.value);
  };

  const onLogout = logout;
  const logInOutKey = isAuthenticated ? t('logout') : t('title');
  const authFn = isAuthenticated ? onLogout : onLogin;

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.SIGNUP.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>
          {isAuthenticated ? (
            <Link to={Route.LOGOUT.path}>{t('logout')}</Link>
          ) : (
            <>
              <form onSubmit={e => e.preventDefault()}>
                <div>
                  <input
                    name="username"
                    onChange={onUsernameChange}
                    placeholder="username"
                    type="text"
                    value={username}
                  />
                </div>
                <div>
                  <input
                    name="password"
                    onChange={onPasswordChange}
                    placeholder="password"
                    type="password"
                    value={password}
                  />
                </div>
                <Button
                  context={PRIMARY}
                  disabled={isDisabled}
                  onClick={authFn}
                  text={logInOutKey}
                  type="submit"
                />
              </form>
              <br />
              or <Link to={Route.SIGNUP.path}>{t('signup')}</Link>
            </>
          )}
        </Body>
      </Content>
    </Page>
  );
};

export { Login };
