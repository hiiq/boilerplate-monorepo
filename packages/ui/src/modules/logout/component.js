import { useHistory } from 'react-router-dom';
import { useAuth } from 'shared/useAuth';
import { useTimeout } from 'shared/useTimeout';
import { Route } from 'types/route';

const Logout = () => {
  const history = useHistory();
  const { logout } = useAuth();

  useTimeout(() => history.push(Route.LOGIN.path), 1);

  logout();

  return null;
};

export { Logout };
