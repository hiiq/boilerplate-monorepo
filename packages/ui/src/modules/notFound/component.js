import { seconds } from '@puttingpoker/common';
import * as Sentry from '@sentry/browser';
import LogRocket from 'logrocket';
import { prop } from 'ramda';
import React, { useEffect, useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { Body, Breadcrumb, Breadcrumbs, Content, Header } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import styled from 'styled-components/macro';
import { LogLevel } from 'types/logLevel';
import { Route } from 'types/route';

const SECONDS_TO_REDIRECT = 15;

const logNotFound = () => {
  Sentry.withScope(scope => {
    scope.setExtra('sessionUrl', LogRocket.sessionURL);
    scope.setExtra('pathname', prop('pathname', window.location));
    scope.setTag('search', prop('search', window.location));
    scope.setLevel(LogLevel.INFO);
    Sentry.captureException(
      new Error('User visited a resource that could not be found')
    );
  });
};

const NotFound = () => {
  const t = useTranslate({
    component: 'notFound',
    namespace: 'notFound',
  });
  const [tics, setTics] = useState(SECONDS_TO_REDIRECT);
  const history = useHistory();
  const location = useLocation();

  useEffect(() => {
    logNotFound();
  }, []);

  useEffect(() => {
    let interval = null;

    if (tics <= 1) return history.goBack();

    const ticToc = () => setTics(tics - 1);

    interval = setInterval(ticToc, seconds(1));

    return () => clearInterval(interval);
  }, [history, tics, setTics]);

  const { pathname } = location;

  const Italicize = styled.span`
    font-style: italic;
    margin-left: 0.25rem;
  `;

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.NOT_FOUND.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>
          <p>
            {t('resourceNotFound')}:<Italicize>{pathname}</Italicize>
          </p>
          <p>{t('returning', { seconds: tics })}</p>
        </Body>
      </Content>
    </Page>
  );
};

export { NotFound };
