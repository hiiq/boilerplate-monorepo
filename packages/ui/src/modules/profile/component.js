import React from 'react';
import { Body, Content, Header, Breadcrumbs, Breadcrumb } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';

const Profile = () => {
  const t = useTranslate({
    component: 'profile',
    namespace: 'profile',
  });

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.USER_PROFILE.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>&nbsp;</Body>
      </Content>
    </Page>
  );
};

export { Profile };
