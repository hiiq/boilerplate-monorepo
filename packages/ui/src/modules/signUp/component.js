import React from 'react';
import { Body, Content, Header, Breadcrumbs, Breadcrumb } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';

const SignUp = () => {
  const t = useTranslate({
    component: 'signUp',
    namespace: 'signUp',
  });

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.SIGN_UP.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body></Body>
      </Content>
    </Page>
  );
};

export { SignUp };
