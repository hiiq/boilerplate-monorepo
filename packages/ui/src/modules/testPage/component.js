const TestPage = () => {
  throw new Error('Throw an error to test the Error Page');
};

export { TestPage };
