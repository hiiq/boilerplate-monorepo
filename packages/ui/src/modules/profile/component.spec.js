import * as Hooks from '@apollo/client';
import React from 'react';
import { Mutation } from 'shared/graphql/mutation';
import { Query } from 'shared/graphql/query';
import { render } from 'testHelpers';
import { Profile } from './component';

jest.mock('shared/Content');
jest.mock('shared/Page');

describe('<Profile/>', () => {
  const defaultProps = {};

  const renderComponent = overrides =>
    render(<Profile {...defaultProps} {...overrides} />);

  beforeEach(() => {
    const useMutation = td.replace(Hooks, 'useMutation');
    const useQuery = td.replace(Hooks, 'useQuery');

    td.when(
      useMutation(Mutation.USER_SELF_UPDATE, {
        refetchQueries: [{ query: Query.ME }],
      })
    ).thenReturn([
      jest.fn().mockName('userSelfUpdate'),
      { data: { userSelfUpdate: true }, loading: false },
    ]);
    td.when(useQuery(Query.ME)).thenReturn({
      data: {
        me: { email: 'EMAIL@EMAIL.EMAIL', id: 'ID', username: 'USERNAME' },
      },
      loading: false,
    });
  });

  test('renders profile page', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });
});
