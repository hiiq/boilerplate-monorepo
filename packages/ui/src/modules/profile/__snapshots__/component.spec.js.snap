// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> renders profile page 1`] = `
<x-page>
  <x-content>
    <x-breadcrumbs>
      <x-breadcrumb
        isend="true"
        text="t(title)"
        to="/profile"
      />
    </x-breadcrumbs>
    <x-header
      title="t(title)"
    />
    <x-body>
      <form>
        <div>
          <input
            name="username"
            placeholder="Username"
            readonly=""
            type="text"
            value="USERNAME"
          />
        </div>
        <div>
          <input
            name="email"
            placeholder="Email"
            type="email"
            value="EMAIL@EMAIL.EMAIL"
          />
        </div>
        <div>
          <button
            aria-disabled="false"
            aria-label="t(updateProfile)"
            class="Button__Container-sc-1v65cga-0 jpjwHS"
            role="button"
            type="submit"
          >
            t(updateProfile)
          </button>
        </div>
      </form>
    </x-body>
  </x-content>
</x-page>
`;
